### PR TITLE
remove xplot dependency

### PR DESCRIPTION
**Pull request by colombod**
Create separate extension package for xplot , addressing #582

### CODE REVIEW

File: samples/notebooks/csharp/Samples/HousingML.dib [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #r "nuget:Microsoft.ML,1.5.1"
 #r "nuget:Microsoft.ML.AutoML,0.17.1"
 #r "nuget:Microsoft.Data.Analysis,0.4.0"
+#r "nuget:XPlot.Plotly.NET.Interactive.3.1.0"
```
Comment (jonsequitur) [inline samples/notebooks/csharp/Samples/HousingML.dib:6]: I think we should drop the "NET" from the package name.

File: src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj [hunk 2]
```diff
@@ -27,11 +27,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Content Include="$(MSBuildThisFileDirectory)Modules\**"
-            Link="Modules\%(RecursiveDir)%(FileName)%(Extension)"
-            PackagePath="contentFiles/any/any/Modules"
-            PackageCopyToOutput="true"
-            CopyToOutputDirectory="PreserveNewest" />
+    <Content Include="$(MSBuildThisFileDirectory)Modules\**" Link="Modules\%(RecursiveDir)%(FileName)%(Extension)" PackagePath="contentFiles/any/any/Modules" PackageCopyToOutput="true" CopyToOutputDirectory="PreserveNewest" />
```
Comment (daxian-dbw) [inline src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj:30]: Given the below `Content` properties are all in multi-line format, maybe you should revert this change?

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs [hunk 3]
```diff
@@ -81,43 +75,59 @@ public PowerShellKernel() : base(DefaultKernelName)
...
+               "Modules");
 
-            Environment.SetEnvironmentVariable(
-                PSModulePathEnvName,
-                $"{psJupyterModulePath}{Path.PathSeparator}{psModulePath}");
+            AddModulePath(psJupyterModulePath);
+
+            // Get the AddAccelerator method
+            var acceleratorType = typeof(PSObject).Assembly.GetType("System.Management.Automation.TypeAccelerators");
+            _addAccelerator = acceleratorType?.GetMethod("Add", new[] { typeof(string), typeof(Type) });
```
Comment (daxian-dbw) [inline src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:106]: `_addAccelerator` can be made a static readonly field, and the initialization can be moved to the static ctor.

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs [hunk 4]
```diff
@@ -81,43 +75,59 @@ public PowerShellKernel() : base(DefaultKernelName)
...
 
             RegisterForDisposal(pwsh);
             return pwsh;
         }
 
+        public void AddModulePath(string modulePath)
```
Comment (daxian-dbw) [inline src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:112]: Is this method (and `AddAccelerator`) used anywhere? I found `KernelExtension.fs` from the other PR has the same implementation.

File: samples/notebooks/csharp/Samples/HousingML.dib (hunk 1 above)
Comment (colombod) [reply samples/notebooks/csharp/Samples/HousingML.dib:6]: done

File: src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj (hunk 2 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj:30]: done

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs (hunk 4 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:112]: the other repo has the duplicated code, with this pr I am planning to move the api on the powershellkernel surface and use it from extensions

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs (hunk 3 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:106]: sounds good

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs [hunk 5]
```diff
@@ -81,43 +80,57 @@ public PowerShellKernel() : base(DefaultKernelName)
...
-                "Modules");
+               Path.GetDirectoryName(typeof(PowerShellKernel).Assembly.Location),
+               "Modules");
 
-            Environment.SetEnvironmentVariable(
-                PSModulePathEnvName,
-                $"{psJupyterModulePath}{Path.PathSeparator}{psModulePath}");
+            AddModulePath(psJupyterModulePath);
+
+
```
Comment (daxian-dbw) [inline src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:109]: Remove extra new lines.

Comment (daxian-dbw) [inline src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:109]: Please remove extra new lines.

**Review by daxian-dbw — APPROVED** [review]
The PowerShell kernel changes LGTM

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs (hunk 4 above)
Comment (daxian-dbw) [reply src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs:115]: sounds good

File: src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs (hunk 5 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs]: done

**Review by jonsequitur — APPROVED** [review]
